python/pyspark/sql/tests/streaming/test_streaming.py

```diff
@@ -392,6 +392,31 @@ def test_streaming_with_temporary_view(self):
                 set([Row(value="view_a"), Row(value="view_b"), Row(value="view_c")]), set(result)
             )
 
+    def test_streaming_drop_duplicate_within_watermark(self):
+        """
+        This verfies dropDuplicatesWithinWatermark works with a streaming dataframe.
+        """
+        user_schema = StructType().add("time", TimestampType()).add("id", "integer")
+        df = (
+            self.spark.readStream.option("sep", ";")
+            .schema(user_schema)
+            .csv("python/test_support/sql/streaming/time")
+        )
+        q1 = (
+            df.withWatermark("time", "2 seconds")
+            .dropDuplicatesWithinWatermark(["id"])
+            .writeStream.outputMode("update")
+            .format("memory")
+            .queryName("q1")
+            .trigger(availableNow=True)
+            .start()
+        )
+        self.assertTrue(q1.isActive)
+        time.sleep(20)
+        q1.stop()
+        result = self.spark.sql("SELECT * FROM q1").collect()
+        self.assertTrue(len(result) >= 6 and len(result) <= 9)
```

Comment:
May I know why there is not a single answer? Why this should be a range?

Reply:
After talking offline, we agree that this test case is just to verify the code works in pyspark, and the expected result is correct theoretically so we can keep it as is.